### PR TITLE
Add PiP support on iOS - Events and Page

### PR DESCRIPTION
**Pull request by 123mpozzi**
Support iOS-related Picture-in-Picture (PiP) events.  
And add a new sample Page which will be used to test the upcoming PiP feature.

### Events
- `PictureInPictureEnter` which is emitted when the player view is about to enter Picture-in-Picture mode.
- `PictureInPictureExit` which is emitted when the player view is about to exit Picture-in-Picture mode
- `PictureInPictureEntered` which is emitted when the player view finishes entering Picture-in-Picture mode.
- `PictureInPictureExited` which is emitted when the player view finishes exiting Picture-in-Picture mode.

### Sample Page
- `PictureInPicture` page to test PiP and show how it can be integrated and used (iOS-only for now)

## Testing

Apply [pipEvents.patch](https://github.com/bitmovin/bitmovin-player-flutter/files/13514765/pipEvents.patch) to force-enable PiP on iOS side.  
Then open the new  **Picture-in-Picture sample page** and check for the added events being fired in the output console when using the native UI button.

### CODE REVIEW

File: example/lib/pages/picture_in_picture.dart [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+      ),
+      body: Column(
+        children: [
+          AspectRatio(
+            aspectRatio: 16 / 9,
+            child: PlayerView(
+              player: _player,
+              key: _playerViewKey,
+              onPictureInPictureEnter: (event) =>
+                  _logger.d('received ${event.runtimeType}: ${event.toJson()}'),
```
[the remark this answers is not in the document]
Comment (123mpozzi) [reply example/lib/pages/picture_in_picture.dart]: Address in 4800e12e5eb3faf7d3a9d9e674d6c127cedfb825. 
Also added an event box listening exclusively to PiP events in b0b956f0476a233039c8126a2312ac7534db1387.

File: example/lib/pages/home.dart [hunk 2]
```diff
@@ -26,7 +27,11 @@ void buildSamples() {
   ];
 
   if (Platform.isIOS) {
-    _samples.add(_Sample('Background Playback', BackgroundPlayback.routeName));
+    _samples
+      ..add(_Sample('Background Playback', BackgroundPlayback.routeName))
+      ..add(
+        _Sample('Picture-in-Picture', PictureInPicture.routeName),
+      );
```
Comment (hawk23) [inline example/lib/pages/home.dart:34]: Nit
```suggestion
      ..add(_Sample('Picture-in-Picture', PictureInPicture.routeName));
```

File: example/lib/pages/picture_in_picture.dart [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+  );
+  final _player = Player(
+    config: const PlayerConfig(
+      key: Env.bitmovinPlayerLicenseKey,
+      remoteControlConfig: RemoteControlConfig(isCastEnabled: false),
+    ),
+  );
+  final _logger = Logger();
+
+  final _eventsKey = GlobalKey<EventsState>();
```
Comment (hawk23) [inline example/lib/pages/picture_in_picture.dart:31]: Nit: 
```suggestion
  final _logger = Logger();
  final _eventsKey = GlobalKey<EventsState>();
  
```

**Review by hawk23 — APPROVED** [review]
🎉 Nice!

**Pull request merged**